visualization/app/codeCharta/util/algorithm/treeMapLayout/treeMapHelper.ts

```diff
@@ -203,13 +204,91 @@ export function getBuildingColor(node: CodeMapNode, { appSettings, dynamicSettin
 		return mapColors.flat
 	}
 
-	if (metricValue < dynamicSettings.colorRange.from) {
+	const positiveColorRGB = ColorConverter.convertHexToColorObject(mapColors.positive)
+	const neutralColorRGB = ColorConverter.convertHexToColorObject(mapColors.neutral)
+	const negativeColorRGB = ColorConverter.convertHexToColorObject(mapColors.negative)
+
+	const { colorRange, colorMode } = dynamicSettings
+
+	switch (colorMode) {
+		case ColorMode.trueGradient: {
+			return calculateTrueGradient(colorRange, metricValue, positiveColorRGB, neutralColorRGB, negativeColorRGB)
+		}
+
+		case ColorMode.weightedGradient: {
+			return calculateWeightedGradient(mapColors, colorRange, metricValue, positiveColorRGB, neutralColorRGB, negativeColorRGB)
+		}
+
+		case ColorMode.absolute:
```

Comment:
Nit picking: You could just write:
```suggestion
		default: {
```